coverage.c

```diff
@@ -0,0 +1,671 @@
+    while ((c = getopt_long(argc, argv, "o:L:q:Q:hHw:r:b:mM", lopts, NULL)) != -1) {
+            switch (c) {
+                case 1:
+                        if ((required_flags = bam_str2flag(optarg)) < 0) {
+                             fprintf(stderr,"Could not parse --rf %s\n", optarg); return EXIT_FAILURE;
+                         }; break;
+                case 2:
+                        if ((fail_flags = bam_str2flag(optarg)) < 0) {
+                             fprintf(stderr,"Could not parse --ff %s\n", optarg); return EXIT_FAILURE;
+                         }; break;
+                case 'o': opt_output_file = optarg; opt_full_width = false; break;
+                case 'L': opt_min_len = atoi(optarg); break;
+                case 'q': opt_min_baseQ = atoi(optarg); break;
+                case 'Q': opt_min_mapQ = atoi(optarg); break;
+                case 'w': opt_n_bins = atoi(optarg); opt_full_width = false;
+                          opt_print_histogram = true; opt_print_tabular = false; 
+			              break;
+                case 'r': opt_reg = optarg; break;   // parsing a region requires a BAM header (strdup unnecessary)
+                case 'b': opt_file_list = optarg; break;
+                case 'm': opt_print_histogram = true; opt_print_tabular = false; break;
+                case 'M': opt_full_utf = true; 
+                          opt_print_histogram = true; opt_print_tabular = false; 
+                          break;
+                case 'H': opt_print_header = false; break;
+                case 'h': return usage();
+                default:  if (parse_sam_global_opt(c, optarg, lopts, &ga) == 0) break;
+                      /* else fall-through */
+                case '?': 
+                   if (optopt == 'c')
+                      fprintf (stderr, "Option -%c requires an argument.\n", optopt);
+                    else if (isprint (optopt))
+                      fprintf (stderr, "Unknown option `-%c'.\n", optopt);
+                    else
+                      fprintf (stderr, "Unknown option character `\\x%x'.\n", optopt);
+                    return 1;
+            }
+    }
+    if (optind == argc && !opt_file_list) 
+        return usage();
+
+    // output file provided by user
+    if (opt_output_file != NULL && strcmp(opt_output_file,"-")!=0) {
+        file_out = fopen( opt_output_file, "w" );
+        if (file_out == NULL) {
+            print_error_errno("coverage", "Cannot open \"%s\" for writing.", opt_output_file);
+            return EXIT_FAILURE;
+        }
+    }
+
+    if (opt_n_bins <= 0 || opt_full_width) {
+        // get number of columns of terminal
+        const char* ccolumns = getenv("COLUMNS");
+        int columns = 0;
+        if (ccolumns == NULL) {
+#if defined has_windows
+            CONSOLE_SCREEN_BUFFER_INFO csbi;
+            GetConsoleScreenBufferInfo(GetStdHandle(STD_OUTPUT_HANDLE), &csbi);
+	    columns = csbi.srWindow.Right - csbi.srWindow.Left + 1;
```

Comment:
You still need a default value for columns (80 works fine) for MINGW consoles, as `GetConsoleScreenBufferInfo` doesn't return the correct values in this case and the output looks like

![coverage](https://user-images.githubusercontent.com/6553979/53016434-8decb980-3445-11e9-8215-6826925a31d8.png)

Reply:
Checking the return status of `GetConsoleScreenBufferInfo` now, should capture this case.

Reply:
Let me know if there are still isues